base/sparse/sparsematrix.jl

```diff
@@ -172,22 +172,6 @@ function reinterpret{T,Tv,Ti,N}(::Type{T}, a::SparseMatrixCSC{Tv,Ti}, dims::NTup
     return SparseMatrixCSC(mS, nS, colptr, rowval, nzval)
 end
 
-function reshape{Tv,Ti}(a::SparseMatrixCSC{Tv,Ti}, dims::NTuple{2,Int})
-    if prod(dims) != length(a)
-        throw(DimensionMismatch("new dimensions $(dims) must be consistent with array size $(length(a))"))
-    end
-    mS,nS = dims
-    mA,nA = size(a)
-    numnz = nnz(a)
-    colptr = Array(Ti, nS+1)
-    rowval = similar(a.rowval)
-    nzval = copy(a.nzval)
-
-    sparse_compute_reshaped_colptr_and_rowval(colptr, rowval, mS, nS, a.colptr, a.rowval, mA, nA)
-
-    return SparseMatrixCSC(mS, nS, colptr, rowval, nzval)
-end
-
```

Comment:
This is akin to the sparse views problem.  It's inconsistent, but I think you should restore this to the copying method for now.  Making reshape default to a view is still a huge improvement over the status quo, even if we don't get to 100% consistency.

It may not be all that terrible to just have sparse reshapes and slices always return copies as an exception to the rule.  Heck, NumPy gets away with their fancy-indexing-returns-a-copy, but I have no idea how error prone that is in practice.